specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2023-11-15/restorable.json

```diff
@@ -0,0 +1,2074 @@
+          "description": "The instance id of the regional restorable account."
+        },
+        "creationTime": {
+          "type": "string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "The creation time of the regional restorable database account (ISO-8601 format)."
+        },
+        "deletionTime": {
+          "type": "string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "The time at which the regional restorable database account has been deleted (ISO-8601 format)."
+        }
+      }
+    },
+    "RestorableSqlDatabasesListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RestorableSqlDatabaseGetResult"
+          },
+          "description": "List of SQL database events and their properties."
+        }
+      },
+      "description": "The List operation response, that contains the SQL database events and their properties."
+    },
+    "RestorableSqlDatabaseGetResult": {
+      "description": "An Azure Cosmos DB SQL database event",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "description": "The properties of a SQL database event.",
+          "$ref": "#/definitions/RestorableSqlDatabaseProperties"
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The unique resource Identifier of the ARM resource."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the ARM resource."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The type of Azure resource."
+        }
+      }
+    },
+    "RestorableSqlDatabaseProperties": {
+      "description": "The properties of an Azure Cosmos DB SQL database event",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "type": "object",
+          "description": "The resource of an Azure Cosmos DB SQL database event",
+          "properties": {
+            "_rid": {
+              "type": "string",
+              "readOnly": true,
+              "description": "A system generated property. A unique identifier."
+            },
+            "operationType": {
+              "type": "string",
+              "readOnly": true,
+              "description": "The operation type of this database event.",
+              "$ref": "#/definitions/OperationType"
+            },
+            "canUndelete": {
+              "type": "string",
+              "readOnly": true,
+              "description": "A state of this database to identify if this database is restorable in same account."
+            },
+            "canUndeleteReason": {
```

Comment:
>canUndeleteReason [](http://example.com/codeflow?start=13&length=17)

the name should be unrestorableReason?

Reply:
this name is already part of management changes and being used by portal.
canUndeleteReason refers to the reason because of which a collection cannot be undeleted. (undelete is slightly different from restorable, as it means that user can simply undo their recent delete)